spec/salesforce_bulk/job_spec.rb

```diff
@@ -0,0 +1,247 @@
+ <totalProcessingTime>0</totalProcessingTime>
+ <apiActiveProcessingTime>0</apiActiveProcessingTime>
+ <apexProcessingTime>0</apexProcessingTime>
+</jobInfo>')
+
+      instance = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      actual_job_id = instance.create_job
+      expect(actual_job_id).to eq(job_id)
+      expect(instance.instance_variable_get(:@@job_id)).to eq(job_id)
+
+    end
+  end
+
+  # https://developer.salesforce.com/docs/atlas.en-us.196.0.api_asynch.meta/api_asynch/asynch_api_jobs_close.htm
+  describe '#close_job' do
+    it 'can adhere to the contract with Salesforce' do
+      instance = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      described_class.class_variable_set(:@@job_id, job_id)
+
+      allow(connection).to receive(:post_xml).with(nil, "job/#{job_id}", '<?xml version="1.0" encoding="utf-8" ?><jobInfo xmlns="http://www.force.com/2009/06/asyncapi/dataload"><state>Closed</state></jobInfo>', xml_headers).and_return('<?xml version="1.0" encoding="UTF-8"?><jobInfo
+   xmlns="http://www.force.com/2009/06/asyncapi/dataload">
+ <id>' + job_id + '</id>
+ <operation>' + operation + '</operation>
+ <object>' + sobject + '</object>
+ <createdById>005PJ000002voWnYAI</createdById>
+ <createdDate>2025-10-17T16:42:34.000Z</createdDate>
+ <systemModstamp>2025-10-17T16:42:34.000Z</systemModstamp>
+ <state>Closed</state>
+ <concurrencyMode>Parallel</concurrencyMode>
+ <contentType>CSV</contentType>
+ <numberBatchesQueued>0</numberBatchesQueued>
+ <numberBatchesInProgress>0</numberBatchesInProgress>
+ <numberBatchesCompleted>1</numberBatchesCompleted>
+ <numberBatchesFailed>0</numberBatchesFailed>
+ <numberBatchesTotal>1</numberBatchesTotal>
+ <numberRecordsProcessed>10</numberRecordsProcessed>
+ <numberRetries>0</numberRetries>
+ <apiVersion>34.0</apiVersion>
+ <numberRecordsFailed>0</numberRecordsFailed>
+ <totalProcessingTime>0</totalProcessingTime>
+ <apiActiveProcessingTime>0</apiActiveProcessingTime>
+ <apexProcessingTime>0</apexProcessingTime>
+</jobInfo>')
+
+      expect(described_class.new(operation, sobject, records_or_query, external_field, connection).close_job).to eq({"apexProcessingTime" => ["0"], "apiActiveProcessingTime" => ["0"], "apiVersion" => [version], "concurrencyMode" => ["Parallel"], "contentType" => ["CSV"], "createdById" => ["005PJ000002voWnYAI"], "createdDate" => ["2025-10-17T16:42:34.000Z"], "id" => ["some-job-id"], "numberBatchesCompleted" => ["1"], "numberBatchesFailed" => ["0"], "numberBatchesInProgress" => ["0"], "numberBatchesQueued" => ["0"], "numberBatchesTotal" => ["1"], "numberRecordsFailed" => ["0"], "numberRecordsProcessed" => ["10"], "numberRetries" => ["0"], "object" => ["Lead"], "operation" => ["query"], "state" => ["Closed"], "systemModstamp" => ["2025-10-17T16:42:34.000Z"], "totalProcessingTime" => ["0"], "xmlns" => "http://www.force.com/2009/06/asyncapi/dataload"})
```

Comment:
Why not use `instance` (or call a declared `subject`)?
```suggestion
      expect(subject.close_job).to
```
I saw this in a few other tests as well.